### PR TITLE
fix(asset): fix assets input sanitization for query builders

### PR DESCRIPTION
**Pull request by kkerezsi**
# Pull Request

## 🤨 Rationale

Based on Threat Analysis, we identified that the query builder doesn't sanitize the text inputs and Grafana is susceptible to XSS attacks via dashboard share feature.

## 👩‍💻 Implementation

Used 'xss' lib wrapper to ensure we are not rendering html/script elements in the text inputs

## 🧪 Testing

- Unit testing
- Manual testing

## ✅ Checklist

- [x] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

**Review by Catalin-web — APPROVED** [review]
I see that notebook have a query builder too. Do you know if it is actually used?

**Comment by kkerezsi** [comment]
@Catalin-web I want to do that in a separate PR since the Notebook implementation is slightly different.

**Review by saradei-ni — APPROVED** [review]

File: src/datasources/asset/components/editors/calibration-forecast/CalibrationForecastEditor.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function CalibrationForecastEditor({ query, handleQueryChange, datasource
 
   function onParameterChange(ev: CustomEvent) {
     if (query.filter !== ev.detail.linq) {
-      query.filter = ev.detail.linq;
+      query.filter = filterXSSLINQExpression(ev.detail.linq);
```
Comment (mure) [inline src/datasources/asset/components/editors/calibration-forecast/CalibrationForecastEditor.tsx:73]: Why do we need to do this here if we're already sanitizing inside the query builder component?

**Review by mure — APPROVED** [review]

File: src/datasources/asset/components/editors/calibration-forecast/query-builder/CalibrationForecastQueryBuilder.test.tsx [hunk 2]
```diff
@@ -50,5 +50,12 @@ describe('CalibrationForecastQueryBuilder', () => {
       expect(conditionsContainer?.length).toBe(1);
       expect(conditionsContainer.item(0)?.textContent).toContain(globalVariableOption.label);
     });
+
+    it('should select a sanitized ModelName in query builder', () => {
+      const { conditionsContainer } = renderElement([workspace], [system], 'ModelName = "<script>alert(\'ModelName\')</script>"');
+
+      expect(conditionsContainer?.length).toBe(1);
+      expect(conditionsContainer.item(0)?.innerHTML).not.toContain('alert(\'ModelName\')');
```
Comment (CiprianAnton) [inline src/datasources/asset/components/editors/calibration-forecast/query-builder/CalibrationForecastQueryBuilder.test.tsx:58]: Is it possible to have an exact quality here? Like expect(conditionsContainer.item(0)?.innerHTML) == ?

**Review by CiprianAnton — APPROVED** [review]

File: src/datasources/asset/components/editors/calibration-forecast/query-builder/CalibrationForecastQueryBuilder.test.tsx (hunk 2 above)
Comment (kkerezsi) [reply src/datasources/asset/components/editors/calibration-forecast/query-builder/CalibrationForecastQueryBuilder.test.tsx:58]: Not really. We are comparing HTML elements that are generated via the framework. I would keep as it is . We want to make sure we sanitized any alerts and scrips in the HTML.

**Pull request merged**